libmozdata/connection.py

```diff
@@ -97,6 +97,9 @@ def __init__(self, base_url, queries=None, **kwargs):
             if "raise_error" in kwargs:
                 self.RAISE_ERROR = kwargs["raise_error"]
 
+        if not self.USER_AGENT:
+            config.get("User-Agent", "name", required=True)
```

Comment:
Why was it needed to be moved here?

Reply:
Perhaps there was another / a better way, but unless I did this a bunch of tests failed because in `bugzilla.py` there is `from .connection import Connection, Query` and with the `config.get` in the `Connection` definition, this caused a failure before I could mock anything / use `set_config`, even putting it in the tests init file -- because the failure happened at the module imports stage.

```
ImportError: Failed to import test module: test_bugzilla
Traceback (most recent call last):
  File "/usr/lib/python3.10/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.10/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/mbonhomme/dev/mozilla/libmozdata/tests/test_bugzilla.py", line 14, in <module>
    from libmozdata import bugzilla, handler
  File "/home/mbonhomme/dev/mozilla/libmozdata/libmozdata/bugzilla.py", line 15, in <module>
    from .connection import Connection, Query
  File "/home/mbonhomme/dev/mozilla/libmozdata/libmozdata/connection.py", line 48, in <module>
    class Connection(object):
  File "/home/mbonhomme/dev/mozilla/libmozdata/libmozdata/connection.py", line 58, in Connection
    USER_AGENT = config.get("User-Agent", "name", required=True)
  File "/home/mbonhomme/dev/mozilla/libmozdata/libmozdata/config.py", line 70, in get
    assert value, f"Option {option} in section {section} is not set"
AssertionError: Option name in section User-Agent is not set
```

Reply:
I wonder if we could use [setUpModule](https://docs.python.org/3/library/unittest.html#setupmodule-and-teardownmodule) for this...
It isn't worth spending too much time on it though, so let's keep it like you did.